src/platform/react-hooks/src/hooks/usePresence.ts

```diff
@@ -44,19 +59,23 @@ export function usePresence<T = any>(
 
   const onUnmount = useCallback(() => {
     // if connection is in one of inactive states, leave call will produce exception
-    if (channel.state === 'attached' && !INACTIVE_CONNECTION_STATES.includes(ably.connection.state)) {
+    if (channel.state === 'attached' && !INACTIVE_CONNECTION_STATES.includes(connectionState)) {
       channel.presence.leave();
     }
-  }, [channel, ably.connection.state]);
+  }, [channel, connectionState]);
 
   useEffect(() => {
+    if (INACTIVE_CONNECTION_STATES.includes(connectionState)) {
```

Comment:
I would change this into:

```jsx
const shouldNotEnterPresence = INACTIVE_CONNECTION_STATES.includes(connectionState) || skip;

useEffect(() => {
   if (shouldNotEnterPresence) return;

   await channel.presence.enter(messageOrPresenceObjectRef.current);
   
   return () => {
      if (channel.state === 'attached' && !INACTIVE_CONNECTION_STATES.includes(connectionState)) {
        channel.presence.leave();
      }
   };
}, [shouldNotEnterPresence]);

```

this way we invoke effect only then it's necessary

Reply:
I liked the idea, thank you! Implemented it in https://github.com/ably/ably-js/compare/2743834c0521378015334b8d6af3de4b47058e03..2f9fadd5f8dc7fbc840e2e56f98b3340c721b808.

It works as expected, no "Connection closed" errors and now I finally can't see a problem with the solution. I hope that was the last change and we would be able to merge now.